install glib:x64-linux-dynamic fails

**Host Environment**
- OS: [Ubuntu Linux]
- Compiler: gcc 9.3.0
**To Reproduce**
Steps to reproduce the behavior:

# https://vcpkg.readthedocs.io/en/latest/examples/overlay-triplets-linux-dynamic/
 ./vcpkg/vcpkg install glib:x64-linux-dynamic --overlay-triplets=custom-triplets

**Failure logs** 
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:145 (message):
    Command failed: /home/kai/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake --build . --config Release --target install -- -v -j17
    Working Directory: /home/kai/vcpkg/buildtrees/glib/x64-linux-dynamic-rel
    See logs for more information:
      /home/kai/vcpkg/buildtrees/glib/install-x64-linux-dynamic-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:27 (vcpkg_build_cmake)
  ports/glib/portfile.cmake:50 (vcpkg_install_cmake)
  scripts/ports.cmake:136 (include)


Error: Building package glib:x64-linux-dynamic failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: glib:x64-linux-dynamic
  Vcpkg version: 2020.11.12-

**Additional context**
glib static was built successfully

--- a/ports/glib/CMakeLists.txt
+++ b/ports/glib/CMakeLists.txt
@@ -181,7 +181,7 @@ endif()
 # gmodule
 add_library(gmodule gmodule/gmodule.c)
 target_compile_definitions(gmodule PRIVATE G_LOG_DOMAIN="GModule")
-target_link_libraries(gmodule PRIVATE glib ${Intl_LIBRARIES})
+target_link_libraries(gmodule PRIVATE glib ${CMAKE_DL_LIBS} ${Intl_LIBRARIES})
 target_include_directories(gmodule PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/gmodule>)
 target_include_directories(gmodule PRIVATE ${Intl_INCLUDE_DIRS} PUBLIC $<INSTALL_INTERFACE:include>)
 list(APPEND GLIB_TARGETS gmodule)

--- a/ports/glib/CONTROL
+++ b/ports/glib/CONTROL
@@ -1,6 +1,6 @@
 Source: glib
 Version: 2.52.3
-Port-Version: 25
+Port-Version: 26
 Homepage: https://developer.gnome.org/glib/
 Description: Portable, general-purpose utility library.
 Build-Depends: zlib, pcre, libffi, gettext, libiconv

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2246,7 +2246,7 @@
     },
     "glib": {
       "baseline": "2.52.3",
-      "port-version": 25
+      "port-version": 26
     },
     "glibmm": {
       "baseline": "2.52.1",

--- a/versions/g-/glib.json
+++ b/versions/g-/glib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "4ae9d07c61fa6787bacd9402f61b416843e36867",
+      "version-string": "2.52.3",
+      "port-version": 26
+    },
+    {
       "git-tree": "5cadd8fd83d64bc11b551994268c796eabf629b6",
       "version-string": "2.52.3",
       "port-version": 25

